ports/esp32/mphalport.c

```diff
@@ -78,6 +84,24 @@ void check_esp_err(esp_err_t code) {
         o_str->data = (const byte *)esp_err_to_name(code); // esp_err_to_name ret's ptr to const str
         o_str->len = strlen((char *)o_str->data);
         o_str->hash = qstr_compute_hash(o_str->data, o_str->len);
+
+        #if MICROPY_ERROR_REPORTING > MICROPY_ERROR_REPORTING_NORMAL
+        mp_printf(MP_PYTHON_PRINTER, "Exception in function '%s' at line %d in file '%s'\n", func, line, file);
+        char err_msg[64];
+        esp_err_to_name_r(code, err_msg, sizeof(err_msg));
+        char str[256];
+        snprintf(str, sizeof(str), MP_ERROR_TEXT("0x%04X, in function '%s' at line %d in file '%s'"), code, func, line, file);
+        mp_obj_str_t *o_str2 = m_new_obj_maybe(mp_obj_str_t);
+        if (o_str2 != NULL) {
+            o_str2->base.type = &mp_type_str;
+            o_str2->data = (const byte *)str;
+            o_str2->len = strlen((char *)o_str2->data);
+            o_str2->hash = qstr_compute_hash(o_str2->data, o_str2->len);
+            // raise
+            mp_obj_t args[3] = { MP_OBJ_NEW_SMALL_INT(pcode), MP_OBJ_FROM_PTR(o_str), MP_OBJ_FROM_PTR(o_str2)};
```

Comment:
I suggest to just have two arguments, the pcode and then an error message string.